src/lib/cpd/CpdEngine.ts

```diff
@@ -190,7 +190,7 @@ export class CpdEngine extends AbstractRuleEngine {
 			const cpdXml = stdout.slice(xmlStart, xmlEnd + cpdEnd.length);
 			const cpdJson: Element = xml2js(cpdXml, {compact: false, ignoreDeclaration: true}) as Element;
 
-			const duplications =  cpdJson.elements[0].elements;
+			const duplications = (cpdJson.elements[0].elements || []).filter(element => element.name === "duplication");
```

Comment:
Could you add a comment on why this is required?

Reply:
For some reason, v6.48.0 of PMD now adds extra elements to the CPD XML output in addition to the duplication elements. So we needed to filter those out.
I can add a comment in the code as well if you think it's necessary.